app/Http/Controllers/UploadController.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\File;
+
+class UploadController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth')->only('upload');
+    }
+
+    public function upload(Request $request)
+    {
+        if (!$request->user()->canUpload()) {
+            return response(['success' => false, 'reason' => 'User has no upload privilege' ], 403);
+        }
+
+        $request->validate([
+            'mismatchFile' => [ 'required', 'file', 'max:' . env('UPLOAD_SIZE_LIMIT'), 'mimes:csv,txt' ]
+        ]);
+
+        $uploadName = $request->name;
+        $description = $request->description;
+        $filename = date('Ymd_His') . '-mismatch-upload.' . $request->user()->mw_userid . '.csv';
+        $request->file('mismatchFile')->storeAs('mismatch-files', $filename);
```

Comment:
(Probably for a follow-up:) We should mount this into the container as a volume, via docker-compose.

Reply:
To be honest, we don't even know what we are going to do with the uploaded file post import. Maybe we will just delete it?

Reply:
Could be useful for debugging purposes, but maybe I'm over engineering this already.